api/nodes/views.py

```diff
@@ -232,12 +236,38 @@ def get_default_odm_query(self):
         query = base_query & permission_query
         return query
 
-    # overrides ListCreateAPIView
+    # overrides ListBulkCreateJSONAPIView, BulkUpdateJSONAPIView
     def get_queryset(self):
-        query = self.get_query_from_request()
-        return Node.find(query)
+        # For bulk requests, queryset is formed from request body.
+        if is_bulk_request(self.request):
+            query = Q('_id', 'in', [node['id'] for node in self.request.data])
 
-    # overrides ListCreateAPIView
+            user = self.request.user
+            if user.is_anonymous():
+                auth = Auth(None)
+            else:
+                auth = Auth(user)
+
+            nodes = Node.find(query)
+            for node in nodes:
+                if not node.can_edit(auth):
+                    raise PermissionDenied
+            return nodes
+        else:
+            query = self.get_query_from_request()
+            return Node.find(query)
+
+    # overrides ListBulkCreateJSONAPIView, BulkUpdateJSONAPIView, BulkDestroyJSONAPIView
+    def get_serializer_class(self):
+        """
+        Use NodeDetailSerializer which requires 'id'
+        """
+        if self.request.method == 'PUT' or self.request.method == 'PATCH' or self.request.method == 'DELETE':
```

Comment:
This can expressed more concisely as `if self.request.method in ('PUT', 'PATCH', 'DELETE')`